homeassistant/components/cookidoo/config_flow.py

```diff
@@ -0,0 +1,211 @@
+"""Config flow for Cookidoo integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import TYPE_CHECKING, Any
+
+from cookidoo_api import (
+    DEFAULT_COOKIDOO_CONFIG,
+    Cookidoo,
+    CookidooAuthException,
+    CookidooLocalizationConfig,
+    CookidooRequestException,
+    get_localization_options,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_NAME, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_LOCALIZATION, DOMAIN, LOCALIZATION_SPLIT_CHAR
+from .coordinator import CookidooConfigEntry
+from .helpers import cookidoo_localization_for_key
+
+_LOGGER = logging.getLogger(__name__)
+
+AUTH_DATA_SCHEMA = {
+    vol.Required(CONF_EMAIL): TextSelector(
+        TextSelectorConfig(
+            type=TextSelectorType.EMAIL,
+            autocomplete="email",
+        ),
+    ),
+    vol.Required(CONF_PASSWORD): TextSelector(
+        TextSelectorConfig(
+            type=TextSelectorType.PASSWORD,
+            autocomplete="current-password",
+        ),
+    ),
+}
+
+
+class CookidooConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Cookidoo."""
+
+    def __init__(self) -> None:
+        """Init config flow."""
+        super().__init__()
+        self.reauth_entry: CookidooConfigEntry | None = None
+        self.reconfigure_entry: CookidooConfigEntry | None = None
+        self.LOCALIZATION_DATA_SCHEMA: dict | None = None
+        self.localizations: list[CookidooLocalizationConfig] = []
```

Comment:
actually, I talked to someone from core and changed my mind: Let's move them out of `__init__` again, will save the type checking (sorry 😅)

Reply:
@zweckj Dont worry :)